aspnetcore/log-mon/metrics/metrics/samples/metric-tests/BasicTests.cs

```diff
@@ -0,0 +1,38 @@
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.Metrics;
+using Microsoft.VisualStudio.TestPlatform.TestHost;
+using Microsoft.Extensions.Telemetry.Testing.Metering;
+
+// <snippet_TestClass>
+public class BasicTests : IClassFixture<WebApplicationFactory<Program>>
```

Comment:
I moved this class into a sample, but I wasn't able to run the test.
I received the following error, I did not find a solution for it.

```
System.TypeInitializationException: 'The type initializer for 'Microsoft.Extensions.Logging.EventSource.LoggingEventSource' threw an exception.'
TypeLoadException: Could not load type 'System.Runtime.CompilerServices.NullableContextAttribute' from assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
```

Reply:
What version of .NET 8 do you have installed? Maybe the issue would be resolved by using a .NET 8 daily build? https://github.com/dotnet/installer/blob/main/README.md#table

Reply:
Doh. Thanks for the tip, I should've thought about that.
It works after I updated to the latest version `8.0.100-preview.7.23376.3` (I also updated the NuGet packages just to be sure).
See the last commit (da144dee7599b452fe2c9f756ecbed4fdf9ae831) for the changes, in which I had to update the test to turn it green.